### PR TITLE
ctb, testmig: Add migration test script, migration dictator cleanups

### PR DESCRIPTION
**Pull request by mslipper**
Adds a Python script to test the migration end-to-end. Currently, it will:

1. Download a DTL and L2Geth archive from GCS
2. Spin up a forked L1
3. Set state slots so that the deployer owns the legacy system
4. Perform the migration

Also includes some fixes to make the migration dictator work, there were a couple of small changes that still needed to be implemented.

### CODE REVIEW

File: packages/contracts-bedrock/deploy/008-L1StandardBridgeImpl.ts [hunk 1]
```diff
@@ -10,8 +10,9 @@ import {
 const deployFn: DeployFunction = async (hre) => {
   const L1CrossDomainMessengerProxy = await getContractFromArtifact(
     hre,
-    'L1CrossDomainMessengerProxy'
+    'Proxy__OVM_L1CrossDomainMessenger'
   )
+  console.log(L1CrossDomainMessengerProxy.address)
```
Comment (tynes) [inline packages/contracts-bedrock/deploy/008-L1StandardBridgeImpl.ts:15]: we should delete this logline

Comment (mslipper) [reply packages/contracts-bedrock/deploy/008-L1StandardBridgeImpl.ts]: done

**Review by tynes — APPROVED** [review]

File: packages/contracts-bedrock/deploy-config/mainnet.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (smartcontracts) [inline packages/contracts-bedrock/deploy-config/mainnet.json:1]: Can we name this something like `mainnet-forked`?

File: packages/contracts-bedrock/contracts/deployment/MigrationSystemDictator.sol [hunk 3]
```diff
@@ -123,11 +123,11 @@ contract MigrationSystemDictator is BaseSystemDictator {
             abi.encodeCall(OptimismPortal.initialize, ())
         );
 
-        // Upgrade and initialize the L1CrossDomainMessenger.
-        config.globalConfig.proxyAdmin.upgradeAndCall(
+        // Upgrade the L1CrossDomainMessenger. No initializer because this is
```
Comment (smartcontracts) [inline packages/contracts-bedrock/contracts/deployment/MigrationSystemDictator.sol:126]: cc @tynes and @maurelian I pinged you about this already but I'd like a double-check here.

File: packages/contracts-bedrock/deploy/015-MigrationSystemDictator.ts [hunk 4]
```diff
@@ -31,15 +31,31 @@ const deployFn: DeployFunction = async (hre) => {
...
   await deployAndVerifyAndThen({
     hre,
     name: 'MigrationSystemDictator',
     args: [
       {
         globalConfig: {
           proxyAdmin: await getDeploymentAddress(hre, 'ProxyAdmin'),
-          controller: deployer, // TODO
-          finalOwner: hre.deployConfig.proxyAdminOwner,
+          controller, // TODO
```
Comment (smartcontracts) [inline packages/contracts-bedrock/deploy/015-MigrationSystemDictator.ts:57]: Can remove the TODO here

File: packages/contracts-bedrock/deploy/015-MigrationSystemDictator.ts [hunk 5]
```diff
@@ -187,8 +207,14 @@ const deployFn: DeployFunction = async (hre) => {
       `Please transfer ownership of the L1StandardBridge (proxy) to the MigrationSystemDictator located at: ${MigrationSystemDictator.address}`
     )
   }
+  const noSigner = await getContractFromArtifact(
```
Comment (smartcontracts) [inline packages/contracts-bedrock/deploy/015-MigrationSystemDictator.ts:210]: I'll clean this up in the next PR

File: ops/docker/hardhat/package.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
   },
   "license": "MIT",
   "dependencies": {
-    "dotenv": "^10.0.0",
-    "hardhat": "^2.9.6"
+    "dotenv": "^16.0.0",
+    "hardhat": "^2.12.0"
```
Comment (smartcontracts) [inline ops/docker/hardhat/package.json:10]: CI issues probably caused by bumping hardhat. Need to make sure we use the same version of hardhat everywhere in the monorepo.

Comment (mslipper) [reply ops/docker/hardhat/package.json:10]: Roger, will revert.

File: packages/contracts-bedrock/deploy-config/mainnet.json (hunk 2 above)
Comment (mslipper) [reply packages/contracts-bedrock/deploy-config/mainnet.json:1]: Wouldn't we need to duplicate the deployment to do that? Hardhat deploy config expects the names to all match.

**Pull request merged**